app/src/main/java/com/codesoom/assignment/repositories/TaskRepositoryImpl.java

```diff
@@ -0,0 +1,51 @@
+package com.codesoom.assignment.repositories;
+
+import com.codesoom.assignment.models.Task;
+import org.springframework.stereotype.Repository;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Repository
+public class TaskRepositoryImpl implements TaskRepository {
+
+    private final Map<Long, Task> taskMap = new ConcurrentHashMap<>();
+
+    public Collection<Task> findAllTasks() {
+        return Collections.unmodifiableCollection(taskMap.values());
+    }
+
+    @Override
+    public Optional<Task> findById(Long id) {
+        final Task task = taskMap.get(id);
+        return Optional.ofNullable(task);
+    }
+
+    @Override
+    public Task addTask(Task task) {
+        taskMap.put(task.getId(), task);
+        return task;
+    }
+
+    @Override
+    public Optional<Task> deleteById(Long id) {
+        final Task removedTask = taskMap.remove(id);
+        return Optional.ofNullable(removedTask);
+    }
+
+    @Override
+    public Map<Long, Task> deleteTasks(List<Long> idList) {
```

Comment:
이렇게 하는 것도 고려할 수 있겠습니다.
```suggestion
    public Set<Long, Task> deleteTasks(Set<Long> idList) {
```

Reply:
```java
public Set<Task> deleteTasks(Set<Long> idList) {
```
이런 방식을 말씀하시는 게 맞을까요?

Reply:
네. 맞습니다. 리턴 타입은 제가 오타를 냈네요! 다만 고려할 점들이 몇 가지 더 있어요.

- `idList`: 이제 List 가 아니라 Set을 사용하게 되었습니다.
- 리턴 타입: Set 으로 충분하지 않을 수도 있습니다. 만약 Task가 아주 중요한 것이라면 (예를 들어 금융 기관에서의 데이터 마이그레이션 Task 일 수도 있습니다) 삭제 결과로 좀 더 복잡한 응답을 기대할지도 모릅니다. 각각의 Task가 삭제된 시간과 Task 백업 아이디 같은 것들이 포함된 복잡한 타입 같은 거 말이죠. 이런 경우엔 `DeleteReport` 같은 타입을 만들어 리턴하는 것도 고려할 수 있습니다. 다행히 지금 우리가 하고 있는 과제는 단순한 Todo 이고 이런 작업들이 불필요하므로 `Map`, `Set`또는 삭제된 Task의 카운트를 의미하는 int 정도로도 만족할 수 있습니다. 그러나 복잡하고 중요하고 위험한 정보를 다루게 된다면 이런 식으로도 작업할 수 있다는 것을 기억해두면 좋을 거에요.